### PR TITLE
New build tool.

### PR DESCRIPTION
**Pull request by hugelgupf**
Provides a new build tool to replace bb/bb.go and scripts/ramfs.go with a unified interface.

 - Reduces lzma-compressed bb-initramfs size by 1M to 2.8M :) (we didn't used to strip symbols, courtesy of copy and pasting go build commands all over the place...)

### CODE REVIEW

File: pkg/golang/build.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package golang
```
Comment (prattmic) [inline pkg/golang/build.go]: // Package golang ...

File: pkg/golang/build.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	if r := os.Getenv("GOROOT"); r != "" {
+		env.Root = r
+	} else {
+		env.Root = runtime.GOROOT()
+	}
+	return env
+}
+
+// FindPackage returns the full path to `pkg` according to the context's Gopaths.
+func (c Environ) FindPackage(pkg string) (string, error) {
```
Comment (prattmic) [inline pkg/golang/build.go]: Wouldn't `types.ImporterFrom.ImportFrom` be more robust?

https://godoc.org/go/types#ImporterFrom

I think you're supposed to get it with `importer.For("source", nil).(types.ImporterFrom)`? bleh.

https://godoc.org/go/importer#For

File: pkg/golang/build.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	path, err := c.FindPackage(pkg)
+	if err != nil {
+		return err
+	}
+
+	args := []string{
+		"build",
+		"-a", // Force rebuilding of packages.
+		"-o", binaryPath,
+		"-installsuffix", "cgo",
```
Comment (prattmic) [inline pkg/golang/build.go]: Even for no cgo?

File: pkg/golang/build.go (hunk 1 above)
Comment (prattmic) [reply pkg/golang/build.go]: There are lots of missing comments in this package.

File: pkg/golang/build.go (hunk 2 above)
Comment (prattmic) [reply pkg/golang/build.go]: Argh, I see this is already done in a later commit.

**Comment by hugelgupf** [comment]
There's a working, really gross bb mode here now.

**Comment by rminnich** [comment]
This broke NiChrome but only because of the globbing change, I'll just work around it. It's too much of an improvement not tohave. You've tested with dynamic compilation in a VM or something I assume?

**Comment by hugelgupf** [comment]
source mode (scripts/ramfs.go equivalent) is still broken, but bb mode is good to go. Also rewrote the README and added some (but not enough) documentation to Go source.

**Comment by hugelgupf** [comment]
Source mode now works, too.

**Pull request merged**